subsys/bluetooth/host/gatt.c

```diff
@@ -682,13 +682,14 @@ struct notify_data {
 	int err;
 	u16_t type;
 	const struct bt_gatt_attr *attr;
+	bt_gatt_notify_complete_func_t func;
```

Comment:
union {attr, func, data, len} with indicate_params?
Or possibly reuse indicate_params, it has the same parameters except for a different typedef for the callback func.